pkg/innerring/internal/metachain/contracts/method_objects.go

```diff
@@ -0,0 +1,218 @@
+package contracts
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/interop"
+	"github.com/nspcc-dev/neo-go/pkg/core/interop/runtime"
+	"github.com/nspcc-dev/neo-go/pkg/core/native"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/hash"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+)
+
+func (m MetaData) submitObjectPut(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	const argsNumber = 2
+	if len(args) != argsNumber {
+		panic(fmt.Errorf("unexpected number of args: %d expected, %d given", argsNumber, len(args)))
+	}
+	metaInfoRaw, ok := args[0].Value().([]byte)
+	if !ok {
+		panic(fmt.Errorf("unexpected first argument value: %T expected, %T given", metaInfoRaw, args[0].Value()))
+	}
+	metaInfoSI, err := stackitem.Deserialize(metaInfoRaw)
+	if err != nil {
+		panic(fmt.Errorf("cannot deserialize meta information from byte array: %w", err))
+	}
+	metaInfo, ok := metaInfoSI.Value().([]stackitem.MapElement)
+	if !ok {
+		panic(fmt.Errorf("unexpected deserialized meta information value: expected %T, %T, given", metaInfo, metaInfoSI.Value()))
+	}
+	cID, err := requiredInMap(metaInfo, "cid").TryBytes()
```

Comment:
Maps are ordered, you may access `cid` via index.

Reply:
> an array should be used instead if we rely on the order

Agree.

> mb i have not considered smth? 

No, it's just that I'm not the most experienced SC writer.

So not an issue.